docs/upgrade-notes.asciidoc

```diff
@@ -0,0 +1,1714 @@
+[[breaking-changes-summary]]
+== Upgrade notes
+
+////
+USE THE FOLLOWING TEMPLATE to add entries to this document.
+1. Copy from "[discrete]" to the last "====" included.
+2. Paste and edit the template in the right area section. Most recent entries should be at the top of the section. 
+3. Make sure to edit the anchor ID [[REPO-PR]] with proper values, then add the corresponding entry to the doc link service https://github.com/elastic/kibana/blob/main/packages/kbn-doc-links/src/get_doc_links.ts:
+
+codeId: `${KIBANA_DOCS}breaking-changes-summary.html#REPO-PR` // replace `codeId` with the ID of your choice that you want to call in the Upgrade Assistant code, and replace `REPO-PR` with the corresponding values you edited in this document.
```

Comment:
Not 100% I follow what `codeId` is

Reply:
That's the ID assigned to the link in the doc link service file, that can be called in the code instead of hardcoding the link. I slightly edited the template and added a link to a small doc explaining the docs link service, hopefully this will be clearer.